test/extended/apiserver/api_requests.go

```diff
@@ -13,13 +14,22 @@ import (
 	configv1 "github.com/openshift/api/config/v1"
 	apiserverclientv1 "github.com/openshift/client-go/apiserver/clientset/versioned/typed/apiserver/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
 	"k8s.io/kubernetes/test/e2e/framework"
 	e2eskipper "k8s.io/kubernetes/test/e2e/framework/skipper"
 
-	"github.com/openshift/origin/pkg/test/ginkgo/result"
 	exutil "github.com/openshift/origin/test/extended/util"
 )
 
+const unknownUser = "<unknown>"
+
+var (
+	// allowedUsers is a list of users who are allowed to make API calls to deprecated APIs
+	allowedUsers = sets.NewString("system:serviceaccount:openshift-kube-storage-version-migrator:kube-storage-version-migrator-sa")
+	// allowedUsersRE is a regular expression for a user being added in test/extended/etcd/etcd_test_runner.go
+	allowedUsersRE = regexp.MustCompile(`test-etcd-storage-path\w+`)
```

Comment:
Nit:

```suggestion
	allowedUsersRE = regexp.MustCompile(`^test-etcd-storage-path\w+$`)
```

I suppose it's also safe to assume that the namespace names used to generate the suffix of the username won't contain hyphens?

Reply:
I wouldn't bet on the namespace not containing hyphens, they usually do contain, since the generated parts are usually have a hyphen between the prefix and the generated part. To some we also explicitly add `k8s-` so it's probably better not to take that far going assumption. That's why I didn't make this regexp too strict.